pkg/ansible/controller/status/utils.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// RunningReason - Condition is running
+	RunningReason = "Running"
+	// SuccessfulReason - Condition is running due to reconcile being successful
+	SuccessfulReason = "Successful"
+	// FailedReason - Condition is failed due to ansible failure
+	FailedReason = "Failed"
+	// UnknownFailedReason - Condition is unknown
+	UnknownFailedReason = "Unknown"
+)
+
+const (
+	// RunningMessage - message for running reason.
+	RunningMessage = "Running reconcilation"
+	// SuccessfulMessage - message for successful reason.
+	SuccessfulMessage = "Awaiting next reconcilation"
+)
+
+// NewCondition -  condition
+func NewCondition(condType ConditionType, status v1.ConditionStatus, ansibleStatus *AnsibleStatus, reason, message string) *Condition {
+	return &Condition{
+		Type:               condType,
+		Status:             status,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reason,
+		Message:            message,
+		AnsibleStatus:      ansibleStatus,
+	}
+}
+
+// GetCondition returns the condition with the provided type.
+func GetCondition(status Status, condType ConditionType) *Condition {
```

Comment:
What do you think of making this, plus Set and Remove below, methods on Status?

Reply:
Upstream generally avoids adding methods to the API types, so I think it's generally preferable to use functions. Plus, everything they're messing with is public, so there's not really a reason for it to be a method.